sale_delivery_split_date/models/sale_order_line.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2018 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+from odoo.tools import format_date
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _prepare_procurement_group_vals(self):
+        vals = super()._prepare_procurement_group_vals()
+        if self._get_procurement_group_key()[0] == 24:
+            if self.commitment_date:
+                vals["name"] += "/" + format_date(self.env, self.commitment_date.date())
+        return vals
+
+    def _get_procurement_group_key(self):
+        """Return a key with priority to be used to regroup lines in multiple
+        procurement groups
+        """
+        priority = 24
+        key = super()._get_procurement_group_key()
+        # Check priority
+        if key[0] < priority:
+            if self.commitment_date:
+                # group by date instead of datetime
+                return (priority, self.commitment_date.date())
```

Comment:
@kimirondon 
Hello, my timezone is GMT+7
i create an order having 2 lines:
- 1st line: Delivery date at 07/13/2023 05:00:00
- 2nd line: Delivery date at 07/13/2023 10:00:00
They are in the same date (07/13/2023) but 2 delivery orders are created after confirming the order.
this line `return (priority, self.commitment_date.date())` should take into account the user timezone
![image](https://github.com/OCA/sale-workflow/assets/8724793/4755c7c3-f3fd-453e-9514-fe48232345db)

Reply:
we should add an unittest for this case as well.

Reply:
we can add another commit to improve this later.